gce_image_import_export_tests/test_suites/test_suite_utils.go

```diff
@@ -54,40 +69,150 @@ func TestSuite(ctx context.Context, tswg *sync.WaitGroup, testSuites chan *junit
+	for tt := range testsMap {
+		tts = append(tts, string(tt))
 	}
+	sort.Strings(tts)
+	go func() {
+		for _, ttStr := range tts {
+			tt := TestType(ttStr)
+			m := testsMap[tt]
+			logger.Printf("=== Running TestSuite %v for test type %v ===", testSuiteName, tt)
+
+			var wg sync.WaitGroup
+			for tc, f := range m {
+				wg.Add(1)
+				go func(ctx context.Context, wg *sync.WaitGroup, tc *junitxml.TestCase, tt TestType, f func(
+					context.Context, *junitxml.TestCase, *log.Logger, *testconfig.Project, TestType)) {
+
+					defer wg.Done()
+					if tc.FilterTestCase(regex) {
+						tc.Finish(tests)
+					} else {
+						defer tc.Finish(tests)
+						logger.Printf("Running TestCase %s.%q", tc.Classname, tc.Name)
+						f(ctx, tc, logger, testProjectConfig, tt)
+						logger.Printf("TestCase %s.%q finished in %fs", tc.Classname, tc.Name, tc.Time)
+					}
+				}(ctx, &wg, tc, tt, f)
+			}
+			wg.Wait()
+
+			ttwg.Done()
+			logger.Printf("=== Fnished running TestSuite %v for test type %v ===", testSuiteName, tt)
+		}
+	}()
 
 	go func() {
-		wg.Wait()
+		ttwg.Wait()
 		close(tests)
 	}()
 
 	return tests
 }
 
-// RunCliTool executes cli tool with params
-func RunCliTool(logger *log.Logger, testCase *junitxml.TestCase, cmdString string, args []string) error {
+func runCliTool(logger *log.Logger, testCase *junitxml.TestCase, cmdString string, args []string) error {
 	logger.Printf("[%v] Running command: '%s %s'", testCase.Name, cmdString, strings.Join(args, " "))
-	cmd := exec.Command(fmt.Sprintf("./%s", cmdString), args...)
+	cmd := exec.Command(cmdString, args...)
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	return cmd.Run()
 }
+
+// RunTestCommand runs given test command
+func RunTestCommand(cmd string, args []string, logger *log.Logger, testCase *junitxml.TestCase) bool {
+	if err := runCliTool(logger, testCase, cmd, args); err != nil {
+		logger.Printf("Error running cmd: %v\n", err)
+		testCase.WriteFailure("Error running cmd: %v", err)
+		return false
+	}
+	return true
+}
+
+func auth(logger *log.Logger, testCase *junitxml.TestCase) bool {
+	authLock.Lock()
+	defer authLock.Unlock()
+
+	// This file exists in test env. For local testing, download a creds file from project
+	// compute-image-tools-test.
+	credsPath := "/etc/compute-image-tools-test-service-account/creds.json"
+	cmd := "gcloud"
+	args := []string{"auth", "activate-service-account", "--key-file=" + credsPath}
+	if err := runCliTool(logger, testCase, cmd, args); err != nil {
+		logger.Printf("Error running cmd: %v\n", err)
+		testCase.WriteFailure("Error running cmd: %v", err)
+		return false
+	}
+	return true
+}
+
+func gcloudUpdate(logger *log.Logger, testCase *junitxml.TestCase, latest bool) bool {
+	gcloudUpdateLock.Lock()
+	defer gcloudUpdateLock.Unlock()
+
+	if latest {
+		cmd := "gcloud"
+		args := []string{"components", "repositories", "add",
+			"https://storage.googleapis.com/compute-image-tools-test-gcloud/components-2.json", "--quiet"}
```

Comment:
does this require a separate bucket? we have compute-image-tools-test-resources for things like this

Reply:
The gcloud build script will put a bunch of files/folders to the root of the bucket. Thus, a separate bucket is needed.